src/librustc_macros/src/hash_stable.rs

```diff
@@ -0,0 +1,86 @@
+use synstructure;
+use syn::{self, Meta, NestedMeta};
+use proc_macro2::{self, Ident};
+
+struct Attributes {
+    ignore: bool,
+    project: Option<Ident>,
+}
+
+fn parse_attributes(field: &syn::Field) -> Attributes {
+    let mut attrs = Attributes {
+        ignore: false,
+        project: None,
+    };
+    for attr in &field.attrs {
+        if let Ok(meta) = attr.parse_meta() {
+            if &meta.name().to_string() != "stable_hasher" {
+                continue;
+            }
+            let mut any_attr = false;
+            if let Meta::List(list) = meta {
+                for nested in list.nested.iter() {
+                    if let NestedMeta::Meta(meta) = nested {
+                        if &meta.name().to_string() == "ignore" {
```

Comment:
I think instead of the `any_attr` variable you should match on `meta.name().to_string()` and panic for unknown attributes and also just unwrap all the wrong data that can be in one of the two possible attributes. So no `if let` without an `else` in case there are no convenience functions that give you something unwrappable.

Reply:
There are a ton of `if`s here. It's much simpler to just have a panic at the end if nothing matched.